PSKoans/Koans/Constructs and Patterns/AboutRegularExpressions.Koans.ps1

```diff
@@ -0,0 +1,334 @@
+
+    # ^ $
+    it 'matches the start and end of lines' {
+        <#
+            The ^ symbol matches the start of a line, which is not normally represented in English with a character. The $ represents the end
+            of a line.
+        #>
+
+        $startAndEnd = '__'
+        $startAndEnd -match '^a' | should -BeTrue
+        $startAndEnd -match 'z$' | should -BeTrue
+    }
+
+    # \s \S
+    it 'matches (non-) whitespace characters' {
+        # \s matches whitespace characters (space, tab, etc.) while \S matches anything other than whitespace
+
+        'Room to grow' -match '__' | should -BeTrue
+        '    ' -match '__' | should -BeFalse        # Careful to only fill in a value for the '__', not the '    '
+        'accordian' -match '__' | should -BeFalse
+        'accordian' -match '__' | should -BeTrue
+    }
+
+    # escape
+    it 'escapes other special symbols' {
+        <#
+            As you know now, the backslash is basically a magic regex wand that gives some normal characters special meanings. It does more
+            than that, though. The backslash can also take special meaning away from characters, even itself. The process of removing the
+            special meaning from a letter in regex is called "escaping".
+        #>
+
+        # Escape the period character to match a literal period instead of "any character"
+        'This . character means something else in regex' -match '\.' | should -BeTrue
+        $matches[0] | should -Be '__'
+
+        $dollarValue = 'The price is $4.99.'
+        $dollarMatch = '__'
+        [regex]::Matches($dollarValue, $dollarMatch).Value | should -Be '$4.99'
+
+        $slashPath = '\\path\with\slashes'
+        $slashPattern = '__'
+        [regex]::Matches($slashPath, $slashPattern).Value | should -Be '\slashes'
+    }
+}
+
+describe 'Brackets and Braces' {
+    <#
+        Brackets and braces don't have one overarching purpose in regular expressions. Different types of brackets and braces all mean different
+        things. Sometimes other regex symbols meanings change when they're found inside some brackets or braces.
+    #>
+    
+    # curly braces
+    it 'works like a custom quantifier' {
+        <#
+            Curly braces are effectively customizable quantifiers. The star, plus, and question mark quantifiers from earlier are static in their
+            meaning. Star always means "zero or more", question mark always means "one or zero", and plus always means "one or more". Those cover
+            tons of use cases and examples, but there are plenty more situations where you want to be more specific.
+        #>
+
+        # The portion of this regex in curly braces acts as a quantifier for the symbol that comes immediately before it. It means "exactly 4 digits."
+        'I can count to 1024' -match '\d{4}' | should -BeTrue
+        $matches[0] | should -Be '__'
+
+        '__' -match '\w{10}' | should -BeTrue
+        '__' -match '\w{10}' | should -BeFalse
```

Comment:
Not exactly sure what you have in mind here.

Reply:
Sorry, I mean since we've already introduced `*`, `+`, and `?` it might be a good segue to introduce the curly braces by comparing equivalents to the three preset quantifiers we've already mentioned.

- `?` is the same as `{,1}` or `{0,1}`
- `*` is the same as `{0,}`
- `+` is the same as `{1,}`

Reply:
I'm thinking maybe one example of that with an explanation should be enough. Otherwise the curly braces start to take on too much attention imo.

Reply:
BUT I've added one in the next commit.